drain/test_drain.py

```diff
@@ -0,0 +1,42 @@
+from utils import wait_for
+from pyln.testing.fixtures import *
+
+pluginopt = {'plugin': os.path.join(os.path.dirname(__file__), "drain.py")}
+
+def test_drain(node_factory):
+    # build a simple circular setup to run tests on
+    # l1, l2, l3, l4 = node_factory.line_graph(4, announce_channels=True, wait_for_announce=True, opts=pluginopt)
+    # ValueError: ('Error waiting for {}', <function NodeFactory.line_graph.<locals>.<lambda> at 0x7f299deb67b8>)
+    l1, l2, l3, l4 = node_factory.line_graph(4, announce_channels=True, opts=pluginopt)
+
+    l4.rpc.connect(l1.info['id'], 'localhost', l1.port)
+    scid4 = l4.fund_channel(l1, 10**5)
+    scid1 = l1.get_channel_scid(l2)
+    scid2 = l1.get_channel_scid(l3)
+    scid3 = l1.get_channel_scid(l4)
+
+    # wait for everyone to see each others channels as active
+    #l1.wait_channel_active(l2);
+    wait_for(lambda: [c['active'] for c in l1.rpc.listchannels(scid1)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l1.rpc.listchannels(scid2)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l1.rpc.listchannels(scid3)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l1.rpc.listchannels(scid4)['channels']] == [True, True])
+
+    wait_for(lambda: [c['active'] for c in l2.rpc.listchannels(scid1)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l2.rpc.listchannels(scid2)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l2.rpc.listchannels(scid3)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l2.rpc.listchannels(scid4)['channels']] == [True, True])
+
+    wait_for(lambda: [c['active'] for c in l3.rpc.listchannels(scid1)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l3.rpc.listchannels(scid2)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l3.rpc.listchannels(scid3)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l3.rpc.listchannels(scid4)['channels']] == [True, True])
+
+    wait_for(lambda: [c['active'] for c in l4.rpc.listchannels(scid1)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l4.rpc.listchannels(scid2)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l4.rpc.listchannels(scid3)['channels']] == [True, True])
+    wait_for(lambda: [c['active'] for c in l4.rpc.listchannels(scid4)['channels']] == [True, True])
```

Comment:
You can likely check that `l1.rpc.listchannels() == n.rpc.listchannels()` for all `n in [l2,l3,l4]` and be done with it :-)

Reply:
@cdecker 

thanks :D

Im just messing around... ln tests 'utils' module we have for main repo is not available, so no `wait_for` anyway...

Also I think I discovered a but in linear_graph together with wait_for_announce True..

Reply:
Also, Im not shure what `l1.rpc.listchannels() == n.rpc.listchannels()` actually tests. Is the `listchannels` output sorted equally for all nodes?